tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -246,12 +246,25 @@ private async Task RestorePinnedWidgetsAsync(Widget[] hostWidgets)
         var finalPlace = 0;
         foreach (var orderedWidget in restoredWidgetsWithPosition)

```

Comment:
Can you merge `restoredWidgetsWithPosition` and `restoredWIdgetsWithoutPosition` and use one loop?  Then the loop can be in a for loop and you get `finalPlace`as the loop index.